xc/xenops_server_xen.ml

```diff
@@ -118,7 +118,7 @@ module VmExtra = struct
     shadow_multiplier: float;
     memory_static_max: int64;
     suspend_memory_bytes: int64;
-    qemu_vbds: (Vbd.id * (int * qemu_frontend)) list;
+    qemu_vbds: (Vbd.id * (int * qemu_frontend option)) list;
```

Comment:
@jonludlam would this break RPU or cross-pool migration?

Reply:
I had a more detailed look at the code, and I think this change is fine.

The `VmExtra.non_persistent_t` type that was modified has a `[@@deriving rpc]` at the end but I see no use of `VmExtra.rpc_of_non_persistent_t` or of `VmExtra.non_persistent_t_of_rpc` in the code, or of `VmExtra.rpc_of_t` / `VmExtra.t_of_rpc`, so I don't see how changing this structure would create incompatibilities with the original version in older hosts:
```
repos/xenopsd$ grep -r VmExtra |grep rpc
xl/xenops_server_xenlight.cppo.ml:    } |> VmExtra.rpc_of_persistent_t |> Jsonrpc.to_string
xl/xenops_server_xenlight.cppo.ml:  state.VmExtra.persistent |> VmExtra.rpc_of_persistent_t |> Jsonrpc.to_string
xl/xenops_server_xenlight.cppo.ml:  let persistent = state |> Jsonrpc.of_string |> VmExtra.persistent_t_of_rpc in
xc/xenops_server_xen.ml:    } |> VmExtra.rpc_of_persistent_t |> Jsonrpc.to_string
xc/xenops_server_xen.ml:    state.VmExtra.persistent |> VmExtra.rpc_of_persistent_t |> Jsonrpc.to_string
xc/xenops_server_xen.ml:    let persistent = state |> Jsonrpc.of_string |> VmExtra.persistent_t_of_rpc in
xc/xenops_server_xen.ml:    persistent |> VmExtra.rpc_of_persistent_t |> Jsonrpc.to_string |> fun state_new ->
```